lib/scrabble.rb

```diff
@@ -15,4 +13,81 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+
+  def score(word)
+    if word == nil
+      return 0
+    elsif word.is_a? Integer
+      return 0
+    else
+      data = word.upcase.split("")
+      score_conversion_string(data)
+    end
+  end
+
+  def score_conversion_string(data)
+    if data == []
+      return 0
+    else
+      data.map do |letter|
+        @key[letter]
+      end.inject(:+)
+    end
+  end
+
+
+  def convert_letters_to_array(data)
+    data.map do |letter|
+      @key[letter]
+    end
+  end
+
+
+  def score_with_multipliers(data, multipliers, double = 1)
+    data = data.upcase.split("")
+    if data.count >= 7
+      new_array = convert_letters_to_array(data)
+      new_array.zip(multipliers).map do |x,y|
+        x * y
+      end.inject(:+) + 10 * double
```

Comment:
If you're using Ruby 2.4, you can use `#sum` instead of `#inject`; it makes the code a bit more readable